R/use_lintr.R

```diff
@@ -43,5 +43,21 @@ use_lintr <- function(path = ".", type = c("tidyverse", "full")) {
     )
   )
   write.dcf(the_config, config_file, width = Inf)
+
+  # Check if config_file is in package i.e. lintr_option("linter_file") != "../.lintr"
+  pkg_path <- normalizePath(path, mustWork = FALSE, winslash = "/")
+  if(startsWith(config_file, prefix = pkg_path)) {
+    # Skip a extra character for the leading `/`
+    rel_path <- substring(config_file, first = nchar(pkg_path) + 2L, last = nchar(config_file))
+    ignore_path <- file.path(pkg_path, ".Rbuildignore")
+    if (!file.exists(ignore_path)) file.create(ignore_path)
```

Comment:
This should only run when we are in a package (Check presence of a DESCRIPTION file?)

Reply:
The documentation of the function says:
```R
#' @param path Path to project root, where a `.lintr` file should be created.
#' If the `.lintr` file already exists, an error will be thrown.
```
and in the function we use `lintr_option("linter_file")` to specify the path of the ".lintr" file relative to the project path and the ".lintr" file only makes sense in a package. So I think it would be ok to assume that the path is that of a package.

If you want me to change it, you could also add a parameter "add_to_ignore" which is set to `file.exists(file.path(path, "DESCRIPTION"))` by default.

Reply:
You can use a `.lintr` config for non-package R projects too.
I don't think an argument is necessary, but feel free to add it anyway. AFAIR, `usethis` doesn't provide an option like this, but is transparent about what changes are made.